src/scripts/selectors/UserSelectors.js

```diff
@@ -1,7 +1,21 @@
+/* @flow */
+
 import { createSelector } from 'reselect'
 
-import { getIsLoggingIn, getUser } from 'selectors/index'
+import { getLoginRequestStatus, getUser } from 'selectors/index'
+import User from 'records/UserRecords'
+import { REQUEST_STATUS } from 'constants/ApplicationConstants'
+
+import type { RootState, RequestStatus } from 'types/ApplicationTypes'
+
+export const getIsLoggingIn: (state: RootState) => boolean = createSelector(
+  [getLoginRequestStatus],
+  (loginRequestStatus: RequestStatus): boolean =>
+    loginRequestStatus === REQUEST_STATUS.PENDING
+)
 
-export const getIsLoggedIn = createSelector(
-  [getUser, getIsLoggingIn], (user, isLoggingIn) => !!(user && user.get('email') && !isLoggingIn)
+export const getIsLoggedIn: (state: RootState) => boolean = createSelector(
+  [getUser, getIsLoggingIn],
+  (user: User, isLoggingIn: boolean): boolean =>
+    !!(user.get('email') && !isLoggingIn)
```

Comment:
@jellegerbrandy I am considering changing this logic to be more like `!!user.get('walletKey') && !isLoggingIn`. My understanding is the presence of a wallet is the true indicator that the user is ready to go. Is that correct?

Or perhaps `isLoggedIn` and `hasWallet` (a selector that doesn't exist yet) should be treated separately. Thoughts?